trusted_applet/main.go

```diff
@@ -240,8 +240,10 @@ func main() {
 		}
 		runDHCP(ctx, nicID, fmt.Sprintf("AW-%s", status.Serial), hostname, runWithNetworking)
 	} else {
-		if err := runWithNetworking(ctx); err != nil && err != context.Canceled {
-			klog.Exitf("runWithNetworking: %v", err)
+		for {
+			if err := runWithNetworking(ctx); err != nil && err != context.Canceled {
+				klog.Exitf("runWithNetworking: %v", err)
```

Comment:
Do you want to change that `Exitf` for a `Warningf` too? (I think that would more closely mirror the behaviour under `DHCP` - I don't _think_ we `Exit` on any errors from  `runWithNetworking` there.